src/features/unified-share-modal/UnifiedShareModal.js

```diff
@@ -320,6 +322,28 @@ class UnifiedShareModal extends React.Component<Props, State> {
         });
     };
 
+    onPillCreate = (type: string) => (pills: Array<SelectOptionProp | Contact>) => {
+        // If this is a dropdown select event, we ignore it
+        if (pills.length === 1 && pills[0].id) {
+            return;
+        }
+        const { getContactsByEmail } = this.props;
+
+        if (getContactsByEmail) {
+            const emails = pills.map(pill => pill.value);
+            // $FlowFixMe
+            getContactsByEmail({ emails }).then((contacts: Object) => {
+                if (type === INVITE_COLLABS_CONTACTS_TYPE) {
+                    this.setState({ inviteCollabsContacts: mergeContacts(this.state.inviteCollabsContacts, contacts) });
+                } else if (type === EMAIL_SHARED_LINK_CONTACTS_TYPE) {
+                    this.setState({
+                        emailSharedLinkContacts: mergeContacts(this.state.emailSharedLinkContacts, contacts),
```

Comment:
In both these clauses you should use functions to setState.

```jsx
this.setState(prevState => ({
  emailSharedLinkContacts: mergeContacts(prevState.emailSharedLinkContacts, contacts)
}));
```

Reply:
good idea